tools/browser_compat_data.json

```diff
@@ -0,0 +1,5939 @@
+{
```

Comment:
Can we limit this to just the features we care about.   It should be easy enough to re-generate each time we decide we care about a new feature.

It would also make it easier to eyeball / review.

Reply:
We could but idk seems like an unnecessary overhead to add new features with little benefit (size is already reasonable)? Besides, this way external JS libraries can use `#if caniuse` for features they care about too, even if core doesn't.

Reply:
I mean, the whole point of this feature is to avoid maintenance overhead so you can just do `#if caniuse` for any feature without having to track them somewhere one by one.

If we still need to add their names somewhere each time we want to compile-test some feature, might as well keep copy-pasting browser versions as well manually like before.

Reply:
It seems pretty easy to add your feature to the list and re-run the script json-generating script.

Reply:
I don't think the size is a big issue.    We can just checkin the whole thing.

But its a shame its hard to confirm in the review that the new and old checks are equivalent.    Can you confirm this and/or explain if there are any differences?

Reply:
> But its a shame its hard to confirm in the review that the new and old checks are equivalent.

Do you mean before/after this PR? Yeah, as mentioned in description, there were already some differences in e.g. manual checks assuming that feature is not supported in Safari but it actually is supported nowadays.

That's why I think trusting the database is better - it's pretty much guaranteed to have more up-to-date information.

Also note that I intentionally limited browser compat data to JavaScript builtins (as in, ECMAScript spec things, excluding all DOM APIs) so it won't update as often as you'd think.

It's relatively rare for new JS core APIs to appear, and I'd expect to see changes only when new browser versions are coming out and implement a couple of features at a time. So if one wants, I think it would be still reasonably easy to review.

Reply:
sgtm